### DISCUSSION TITLE
A lot of M Codes are not working - why?

### DISCUSSION THREAD

**Discussion by cwuetscher** [mriscoc/Ender3V2S1#1062] [Q&A]
Hello together,

I installed the current release on my Enders 3V2 4.2.2 for the first time.

Now I wanted to use some M Codes, but a lot of them are not working.

I wanted to deactivate linear advance with M900 - not working
Deactivate stealthchop with M569 - not working
Deactivate Firmware retrace with M209 - not working
Change Motorcurrent for the Extruder with M906 - not working

Now I wonder why the codes are not working. Anyone has an idea why?

**Comment by mriscoc** [comment]
Some G-codes have requirements; you need to read Marlin manual to better understand how a G-code works.
https://marlinfw.org/meta/gcode/

The use of M209 Is not recommended https://marlinfw.org/docs/features/fwretract.html but if it is required for a particular use case, you should enable it in the configuration files and perform your custom build.

And you also need to understand the capabilities of the hardware/printer, for example, the 4.2.2 board don't have M906 configurable stepper drivers, and they work in standalone mode.

The pre-compiled binaries work with the main board in stock condition, if you do a hardware modification you need to compile a firmware version specifically adapted to your hardware changes in your board.
